background/messages/mCreateWindow.ts

```diff
@@ -15,19 +26,80 @@ const handler: PlasmoMessaging.MessageHandler<
   RequestBody,
   ResponseBody
 > = async (req, res) => {
-  console.log(`[background][messaging][window] Request: ${JSON.stringify(req)}`)
-
-  await windows.create({
-    url: createWindowUrl,
-    focused: true,
-    type: "popup",
-    width: 385,
-    height: 720
-  })
+  console.log(
+    `[background][messaging][window] Request: ${JSON.stringify(req, null, 2)}`
+  )
+
+  const createWindowUrl = `${runtime.getURL(
+    "tabs/mCreateWindow.html"
+  )}?origin=${req.body.params?.[0].origin}&account=${req.body.params?.[0]
+    .account}&tabId=${req.sender.tab.id}`
+
+  const window = await createWindowAsync(createWindowUrl)
+  console.log(
+    `[background][messaging][window] window: ${JSON.stringify(window, null, 2)}`
+  )
+
+  //   const tab = await createTabAsync(createWindowUrl)
+  //   console.log(
+  //     `[background][messaging][window] tab: ${JSON.stringify(window, null, 2)}`
+  //   )
 
   res.send({
     out: `Opened: ${createWindowUrl}`
   })
 }
 
+function createWindowAsync(createWindowUrl: string) {
+  return new Promise((resolve, reject) => {
+    let createdWindow = null
+    windows
+      .create({
+        url: createWindowUrl,
+        focused: true,
+        type: "popup",
+        width: 385,
+        height: 720
+      })
+      .then((window) => {
+        createdWindow = window
+      })
+      .catch((error) => {
+        reject(error)
+      })
```

Comment:
Mark the promise callback async, then we could use async/await in it

```suggestion
  return new Promise(async (resolve, reject) => {
    let createdWindow = null
    try {
        createdWindow = await windows.create({
            url: createWindowUrl,
            focused: true,
            type: "popup",
            width: 385,
            height: 720
        })
    } catch (e) {
        reject(e)
    }    
```